### ISSUE TITLE
RequireJS using incorrect baseUrl in developer mode

### ISSUE THREAD

**Issue by vivek-graas** [magento/magento2#19208]
### Preconditions (*)

1.CE 2.3-develop - HEAD (Latest commit 877bffd)
2.Vanilla magento 2.3 installation on Ubuntu 18.04 and MySQL, **Nginx, FPM**
3. Chrome / Firefox browser on Ubuntu desktop

### Steps to reproduce (*)

1. Install Magento 2.3 from source using contributor guidelines from the HEAD
2. Enable developer mode, clear pub/static and cache folders.
3. Load site in the browser. I get 404 errors on all static resources.
4.Note: I don't get the errors if I install from an earlier commit (Nov 1). Git reset hard and flush cache solves the problem
git reset --hard f33c84fbda6

### Expected result (*)

1. Expect basic theme css and js to load without errors

```
var require = {
--
  | "baseUrl": "http://www.m23store.com/static/version1542197613/frontend/Magento/luma/en_US"
  | };

```

### Actual result (*)

The page source shows **/pub/** being appended to baseUrl for loading static assets. So static files get 404 errors. 
See below. However, I am not certain what causes it. Regenerated pub/static files several times. Permissions look ok too.

```
<script>
--
  | var BASE_URL = 'http://www.m23store.com/';
  | var require = {
  | "baseUrl": "http://www.m23store.com/pub/static/version1542191804/frontend/Magento/luma/en_US"
  | };
  | </script>

```
Found a really old issue with the same problem: https://github.com/magento/magento2/issues/2711
Also ref. my earlier bug report: https://github.com/magento/magento2/issues/19167

**Comment by magento-engcom-team** [comment]
Hi @vphalak. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [ ] Summary of the issue
- [ ] Information on your environment
- [ ] Steps to reproduce
- [ ] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@vphalak do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [ ] yes
- [ ] no

**Comment by vivek-graas** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @vphalak. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @vphalak, here is your [Magento instance](http://34.228.235.121/i-19208-2-3-develop/).
Admin access: [http://34.228.235.121/i-19208-2-3-develop//admin](http://34.228.235.121/i-19208-2-3-develop//admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by vivek-graas** [comment]
The test environment launched above also contains /pub/ in the baseUrl. However, there we don't see 404 errors. So this could be specific to nginx / fpm? I am using the vanilla nginx.conf.sample from the cloned repo. I am confused if the baseUrl should contain /pub/ or not.
(Note: a vanilla environment from few commit before, I don't get /pub/.)

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [ ] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [ ] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [ ] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [ ] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 6. Add label `Issue: Confirmed` once verification is complete. 

- [ ] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by ghost** [comment]
Hi @vphalak thank you for you report, this is expected behavior, in all version of magento this url include path pub.

**Comment by edumart16** [comment]
@engcom-backlog-nazar im having the same problem in 2.3.2 version and i see all the related tickets and nothing works for me. I wait for your answer, I didn't want to generate a new issue because maybe something I did wrong.
